e2e/tests/add-to-cart.spec.ts

```diff
@@ -0,0 +1,48 @@
+const assert = require('assert');
+import { expect } from 'playwright/test';
+import { BASE_URL, PRODUCT_NAME, WOMEN_COLLECTIONS_ENDPOINT } from '../constants';
+import { test } from '../fixtures';
+
+test.describe('Add to cart', () => {
+  test.beforeEach('Navigate to the Collections page', async ({ page }) => {
+    await page.goto(WOMEN_COLLECTIONS_ENDPOINT);
+  });
+
+  test('User is able to add product to cart', async ({ page, collectionsPage, shoppingCart, productPage }) => {
+    if (!PRODUCT_NAME) {
+      assert(PRODUCT_NAME, 'PRODUCT_NAME must be defined');
+    }
+
+    await collectionsPage.getProductByName(PRODUCT_NAME).click();
+    await expect(page.getByLabel('Breadcrumb')).toContainText(PRODUCT_NAME);
+
+    const productPrice = await productPage.productPrice().innerText();
+
+    await productPage.addToCartBtn().click();
+    await expect(page.getByText('Shopping cart')).toBeVisible();
+    await expect(shoppingCart.shoppingCartItems()).toHaveCount(1);
+    await expect(shoppingCart.shoppingCartItems()).toContainText(PRODUCT_NAME);
+    await expect(shoppingCart.cartTotalPrice()).toContainText(productPrice);
+  });
+
+  test('Checkout phase', async ({ page, shoppingCart, collectionsPage, productPage }) => {
+    if (!PRODUCT_NAME) {
+      assert(PRODUCT_NAME, 'PRODUCT_NAME must be defined');
+    }
+
+    if (!BASE_URL) {
+      assert(BASE_URL, 'BASE_URL must be defined');
+    }
+
+    await collectionsPage.getProductByName(PRODUCT_NAME).click();
+    await productPage.addToCartBtn().click();
+    await expect(page.getByText('Shopping cart')).toBeVisible();
+    await expect(shoppingCart.shoppingCartItems()).toContainText(PRODUCT_NAME);
+
+    await page.goto('/?shopify_checkout_action=success');
+
+    await page.waitForURL(BASE_URL);
```

Comment:
`PLAYWRIGHT_TEST_BASE_URL` should have internally set the root. I think `page.waitForUrl('/')` or `page.waitForUrl()` will work here. The cases where you need to explicitly define the the base url should be pretty limited.

Reply:
we can use `page.waitForUrl('/')`, because `waitForUrl` method expects at least 1 argument.